### PR TITLE
Fix typos, rename types, and add null_probability benchmark axis for distinct

### PR DESCRIPTION
**Pull request by PointKernel**
## Description

This PR addresses several minor issues discovered while working on #17467:

- Corrected a typo where `RowHasher` should have been `RowEqual`
- Renamed `hash_set_type` to `distinct_set_t`
- Added a `null_probability` benchmark axis for the distinct benchmark, similar to other stream compaction benchmarks

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/stream_compaction/distinct_helpers.hpp [hunk 1]
```diff
@@ -87,8 +87,8 @@ using hash_set_type =
  * @param mr Device memory resource used to allocate the returned vector
  * @return A device_uvector containing the output indices
  */
-template <typename RowHasher>
-rmm::device_uvector<size_type> reduce_by_row(hash_set_type<RowHasher>& set,
+template <typename RowEqual>
```
Comment (mhaseeb123) [inline cpp/src/stream_compaction/distinct_helpers.hpp:90]: Can we please add the missing `@tparam RowEqual ` to the docstring of this function

**Review by mhaseeb123 — APPROVED** [review]
Minor comment but looks good otherwise

File: cpp/benchmarks/stream_compaction/distinct.cpp [hunk 2]
```diff
@@ -65,6 +66,7 @@ using data_type = nvbench::type_list<int32_t, int64_t>;
 NVBENCH_BENCH_TYPES(nvbench_distinct, NVBENCH_TYPE_AXES(data_type))
   .set_name("distinct")
   .set_type_axes_names({"Type"})
+  .add_float64_axis("null_probability", {0.01})
```
Comment (mhaseeb123) [inline cpp/benchmarks/stream_compaction/distinct.cpp:69]: Should we add more values to this axis (for baseline signals) or this is just to modify it at runtime for particular cases?

File: cpp/src/stream_compaction/distinct_helpers.hpp (hunk 1 above)
Comment (PointKernel) [reply cpp/src/stream_compaction/distinct_helpers.hpp:92]: Good catch! Doc added

**Review by vyasr — APPROVED** [review]

**Pull request merged**